### PR TITLE
Adding CatWISE dataset to repo

### PR DESCRIPTION
**Pull request by oliveirafrancofelipe**

### CODE REVIEW

File: xcell/mappers/mapper_CatWISE.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        config - dict
+          {'data_catalog': 'xcell/tests/data/
+          catwise_agns_masked_final_w1lt16p5_alpha.fits',
+           'mask': 'xcell/tests/data/MASKS_exclude_master_final.fits',
+           'mask_name': 'mask_CatWISE'}
+        """
+        self._get_defaults(config)
+        self.file_sourcemask = config.get('mask_sources', None)
+        self.ra_name = 'ra'
+        self.dec_name = 'dec'
```
Comment (damonge) [inline xcell/mappers/mapper_CatWISE.py:20]: do we need to store these two (self.ra_name, self.dec_name) for any reason?

File: xcell/mappers/mapper_CatWISE.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        self.radius_name = 'radius'
```
Comment (damonge) [inline xcell/mappers/mapper_CatWISE.py:21]: same question about `self.radius_name`

File: xcell/mappers/mapper_CatWISE.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+            GLON, GLAT = r(self.cat_data['ra'], self.cat_data['dec'],
+                           lonlat=True)
+            self.cat_data['GLON'] = GLON
+            self.cat_data['GLAT'] = GLAT
+            # Angular and flux conditions
+            self.cat_data = self.cat_data[
+                (self.cat_data['w1'] <
+                 self.config.get('flux_max_W1', 16.4)) &
+                (np.fabs(self.cat_data['GLAT']) >
+                 self.config.get('GLAT_max_deg', 30))]
```
Comment (damonge) [inline xcell/mappers/mapper_CatWISE.py:48]: The catalog seems to be already cut at `GLAT > 30`, so we don't need to include this

File: xcell/tests/test_mapper_CatWISE.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
+import numpy as np
+import xcell as xc
+import healpy as hp
+import os
+from astropy.table import Table
+
+
```
Comment (damonge) [inline xcell/tests/test_mapper_CatWISE.py:7]: general question about the tests: are you ever testing the lines of code that mask the circles around the sources? For some reason coveralls isn't really complaining, but I can't see where that's happening...

**Review by damonge — CHANGES_REQUESTED** [review]
Looks great, good job @oliveirafrancofelipe ! A few minor comments

File: xcell/mappers/mapper_CatWISE.py (hunk 1 above)
Comment (oliveirafrancofelipe) [reply xcell/mappers/mapper_CatWISE.py:20]: actually we don't

File: xcell/mappers/mapper_CatWISE.py (hunk 3 above)
Comment (oliveirafrancofelipe) [reply xcell/mappers/mapper_CatWISE.py:48]: Yes, this is true. The only condition necessary is the flux for W1

File: xcell/tests/test_mapper_CatWISE.py (hunk 4 above)
Comment (oliveirafrancofelipe) [reply xcell/tests/test_mapper_CatWISE.py:7]: No, I didn't write a test for it. In the test for get_mask I've used `config.pop('mask_sources')` so neglecting the holes. Should I include a test for it?

File: xcell/mappers/mapper_CatWISE.py [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    def get_catalog(self):
+        if self.cat_data is None:
+            file_data = self.config['data_catalog']
+            self.cat_data = Table.read(file_data)
+            # Galactic coordinates
+            r = hp.Rotator(coord=['C', 'G'])
+            GLON, GLAT = r(self.cat_data['ra'], self.cat_data['dec'],
+                           lonlat=True)
+            self.cat_data['GLON'] = GLON
+            self.cat_data['GLAT'] = GLAT
```
Comment (damonge) [inline xcell/mappers/mapper_CatWISE.py:39]: Don't calculate galactic coordinates if you're not gonna use them.

File: xcell/tests/test_mapper_CatWISE.py (hunk 4 above)
Comment (damonge) [reply xcell/tests/test_mapper_CatWISE.py:7]: ok, let's move on without this test and circle back if we need it.

**Review by damonge — COMMENTED** [review]
OK, last comment. We should think about what to do regarding the redshift distribution. Can you create a placeholder method that reads:
```
    def get_nz(self, dz=0):
        raise NotImplementedError("No dNdz for CatWISE yet")
```
?

**Review by damonge — APPROVED** [review]
LGTM!

**Pull request merged**